Python/compile.c

```diff
@@ -8181,6 +8213,88 @@ add_checks_for_loads_of_uninitialized_variables(basicblock *entryblock,
     return 0;
 }
 
+static int
+expand_del_noerror(basicblock *entryblock, int none_oparg)
+{
+    for (basicblock *b = entryblock; b != NULL; b = b->b_next) {
+        for (int i = 0; i < b->b_iused; i++) {
+            int opcode = b->b_instr[i].i_opcode;
+            int oparg = b->b_instr[i].i_oparg;
+            basicblock *except = b->b_instr[i].i_except;
+            location loc = b->b_instr[i].i_loc;
+            int del_opcode, store_opcode;
+            switch (opcode) {
+                case DELETE_FAST_NOERROR:
+                    // Already proved this can't raise UnboundLocalError.
+                    b->b_instr[i].i_opcode = DELETE_FAST;
+                    break;
+                case DELETE_FAST_CHECK:
+                    b->b_instr[i].i_opcode = DELETE_FAST;
+                    if (insert_instruction(b, i, &(struct instr) {
+                        .i_opcode = POP_TOP,
+                        .i_oparg = 0,
+                        .i_loc = loc,
+                        .i_except = except,
+                    })) {
+                        return -1;
+                    }
+                    if (insert_instruction(b, i, &(struct instr) {
+                        .i_opcode = LOAD_FAST_CHECK,
+                        .i_oparg = oparg,
+                        .i_loc = loc,
+                        .i_except = except,
+                    })) {
+                        return -1;
+                    }
+                    i += 2;
+                    break;
+                case DELETE_FAST_NOERROR_CHECK:
+                case DELETE_DEREF_NOERROR:
+                case DELETE_GLOBAL_NOERROR:
+                case DELETE_NAME_NOERROR:
+                    switch (opcode) {
+                        case DELETE_FAST_NOERROR_CHECK:
+                            del_opcode = DELETE_FAST;
+                            store_opcode = STORE_FAST;
+                            break;
+                        case DELETE_DEREF_NOERROR:
+                            del_opcode = DELETE_DEREF;
+                            store_opcode = STORE_DEREF;
+                            break;
+                        case DELETE_GLOBAL_NOERROR:
+                            del_opcode = DELETE_GLOBAL;
+                            store_opcode = STORE_GLOBAL;
+                            break;
+                        case DELETE_NAME_NOERROR:
+                            del_opcode = DELETE_NAME;
+                            store_opcode = STORE_NAME;
+                            break;
+                    }
+                    b->b_instr[i].i_opcode = del_opcode;
+                    if (insert_instruction(b, i, &(struct instr) {
```

Comment:
Perhaps we can avoid shifting here:

in compiler_nameop, add a couple of NOPs so that there is space for this expansion. If no expansion happens, the NOPs can be removed by a final remove_redundant_nops().

Reply:
optimize_cfg removes NOPs already, so we'd have to either introduce another pseudo-opcode or shuffle around `assemble()` a bit.

Reply:
We could have a new pseudo instruction called BLANK that is ignored by the assembler (has size 0 and emits nothing).